assignments/array-methods.js

```diff
@@ -54,30 +54,45 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach(item => fullName.push(`${item.first_name} ${item.last_name}`));
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map(item => {
+  let newItem = Object.assign({}, item);
+  newItem.first_name = newItem.first_name.toUpperCase();
+  return newItem;
+});
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(item => item.shirt_size === "L");
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((sum, item) => sum + item.donation, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1: How many runners wear small shirts?
+let numSmallShirts = runners.reduce((sum, item) => item.shirt_size === "S" ? sum + 1 : sum, 0);
+console.log(`Number of people wearing small shirts: ${numSmallShirts}`);
 
-// Problem 2
+// Problem 2: Who are the cheapskates? (criteria for a cheapskate: less than $50 donation)
+let cheapskates = runners.filter(item => item.donation < 50);
+console.log(cheapskates);
 
-// Problem 3
\ No newline at end of file
+// Problem 3: How many companies are at the run?
+let companies = [];
+runners.forEach(item => {
+  if (!companies.includes(item.company_name))
+    companies.push(item.company_name)
+});
```

Comment:
Here is a good example of where object destructuring is nice for brevity since you're accessing the same key multiple times:
```javascript
runners.forEach(({ company_name }) => {
  if (!companies.includes(company_name))
    companies.push(company_name)
});
```